lib/models/watch-detector.js

```diff
@@ -130,55 +109,49 @@ class WatchDetector {
+      case NODE:
+        return this.checkNodeWatcher() && new WatchPreference(NODE);
+      case POLLING:
+        return new WatchPreference(POLLING);
+      case EVENTS:
+        return this.watcherFor(WATCHMAN) || this.watcherFor(NODE);
+      default:
+        this.ui.writeLine(`Unknown watcher type: ${watcherType}`);
+        return null;
+    }
+  }
+
   /*
    * @public
    *
    * Selecting the best watcher is complicated, this method (given a preference)
    * will test and provide the best possible watcher available to the current
    * system and project.
    *
-   *
    *  @method findBestWatcherOption
    *  @input {Object} options
-   *  @returns {Object} watch preference
+   *  @returns {WatchPreference} watch preference
   */
   findBestWatcherOption(options) {
-    let preference = this.extractPreferenceFromOptions(options);
-    let original = options.watcher;
+    let preference = options.watcher;
 
-    if (original && POSSIBLE_WATCHERS.indexOf(original) === -1) {
-      return Promise.reject(new SilentError(`Unknown Watcher: \`${original}\` supported watchers: [${POSSIBLE_WATCHERS.join(', ')}]`));
+    if (preference && POSSIBLE_WATCHERS.indexOf(preference) === -1) {
+      return Promise.reject(new SilentError(`Unknown Watcher: \`${preference}\` supported watchers: [${POSSIBLE_WATCHERS.join(', ')}]`));
     }
 
-    if (preference.watcher === WATCHMAN) {
-      preference = this.checkWatchman();
-    }
+    let bestWatcher = this.watcherFor(preference) || this.watcherFor(WATCHMAN) || this.watcherFor(NODE) || this.watcherFor(POLLING);
 
-    let bestOption = preference;
-    let actual;
-
-    if (bestOption.watcher === NODE) {
-      // although up to this point, we may believe Node is the best watcher
-      // this may not be true because:
-      // * not all platforms that run node, support node.watch
-      // * not all file systems support node watch
-      //
-      let appearsToWork = this.testIfNodeWatcherAppearsToWork();
-      actual = new WatchPreference(appearsToWork ? NODE : POLLING);
+    if (preference && preference !== bestWatcher.watcher) {
+      this.ui.writeLine(`Could not use ${preference} watcher, falling back to ${bestWatcher.watcher}`);
     } else {
-      actual = bestOption;
-    }
-    debug.info('foundBestWatcherOption, preference was: %o, actual: %o', options, actual);
-    if (actual /* if no preference was initial set, don't bother informing the user */ &&
-      original /* if no original was set, the fallback is expected */ &&
-      actual.watcher !== original &&
-
-      // events represents either watcherman or node, but no preference
-      !(original === EVENTS && (actual.watcher === WATCHMAN || actual.watcher === NODE))) {
-      // if there was an initial preference, but we had to fall back inform the user.
-      this.ui.writeLine(`was unable to use: "${original}", fell back to: "${actual.watcher}"`);
+      this.ui.writeLine(`Selected ${bestWatcher.watcher} watcher`);
```

Comment:
It's a bit chatty to always output the selected watcher.  Previously we only wrote something if we had to fall back.  What's the reason for changing this?